std/utf.d

```diff
@@ -3490,19 +3490,26 @@ int impureVariable;
 }
 
 /****************************
- * Iterate an input range of characters by char type C.
+ * Iterate an input range of characters by char type `C` by
+ * encoding the elements of the range.
  *
- * UTF sequences that cannot be converted to UTF-8 are replaced by U+FFFD
- * per "5.22 Best Practice for U+FFFD Substitution" of the Unicode Standard 6.2.
- * Hence byUTF is not symmetric.
+ * UTF sequences that cannot be converted to the specified encoding are
+ * replaced by U+FFFD per "5.22 Best Practice for U+FFFD Substitution"
+ * of the Unicode Standard 6.2. Hence byUTF is not symmetric.
  * This algorithm is lazy, and does not allocate memory.
- * Purity, nothrow, and safety are inferred from the r parameter.
+ * `@nogc`, `pure`-ity, `nothrow`, and `@safe`-ty are inferred from the
+ * `r` parameter.
  *
  * Params:
- *      C = char, wchar, or dchar
+ *      C = `char`, `wchar`, or `dchar`
  *      r = input range of characters, or array of characters
  * Returns:
- *      input range of type C
+ *      If r is a range and not auto-decodable, as defined by
+ *      $(REF isAutodecodableString, std, traits), then a forward range is
+ *      returned if the base range is also a forward range, if not this
+ *      returns an input range of characters. Or, if r is a range and it
+ *      is auto-decodable and `is(ElementEncodingType!typeof(r) == C)`,
+ *      then the range is passed to $(LREF byCodeUnit).
```

Comment:
I think for wrapper range functions, we need a list of standard supported range features, and then just list the ones that are supported, and indicate when they are supported (if argument x has this feature, then result has this feature).

I agree we have very little way to specify what the return type can do, since you can't ddoc it. It is actually one thing I hoped would have worked with the horcrux trick, but the docs don't come out.